src/object/srv_obj.c

```diff
@@ -1507,13 +1507,22 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 			goto post;
 		}
 
-		if (ioc->ioc_coc->sc_props.dcp_csum_enabled) {
+		if (ioc->ioc_coc->sc_props.dcp_csum_enabled &&
+		    daos_csummer_initialized(ioc->ioc_coc->sc_csummer)) {
```

Comment:
Isn't this 'daos_csummer_initialized()' already checked above?

Reply:
Oh, you're right. I guess I was being overly cautious.

Reply:
are you planning to push another patch for this one?